impeller/renderer/backend/vulkan/command_encoder_vk.cc

```diff
@@ -157,10 +157,12 @@ bool CommandEncoderVK::Submit() {
     return false;
   }
 
-  return fence_waiter_->AddFence(
-      std::move(fence), [tracked_objects = std::move(tracked_objects_)] {
-        // Nothing to do, we just drop the tracked objects on the floor.
-      });
+  return fence_waiter_->AddFence(std::move(fence),
+                                 [tracked_objects = std::move(tracked_objects_),
+                                  strong_device = std::move(strong_device)] {
```

Comment:
Why is this capturing a strong reference to the `ContextVK` in the lambda?

`TrackedObjectsVK` holds a weak reference to the `CommandPoolVK`.  That should allow cleanup of the tracked objects if the `CommandPoolVK` and the `ContextVK`/`vk::Device` are still alive (and do nothing if they are already gone)

Reply:
Our options are to keep a strong hold on the device in the fence, or to release the buffer without crashing.  I believe you prefer removing the strong link and calling `release()`.  I think I do too so I did that, although I'm unhappy about the C++ vulkan wrapper and how it looks like a leak.